be/src/exprs/agg/array_agg.h

```diff
@@ -249,6 +248,25 @@ class ArrayAggAggregateFunctionV2
             materialize_column_by_permutation(tmp.get(), {(*state_impl.data_columns)[0]}, perm);
             res = ColumnPtr(std::move(tmp));
         }
+        // further remove duplicated values, pick the last unique one to identify the last sep and don't output it.
+        // TODO(fzh) optimize it later
+        if (ctx->get_is_distinct()) {
+            bool is_duplicated = false;
+            Filter filter(elem_size, 1);
+            for (auto row_id = 0; row_id < elem_size; row_id++) {
```

Comment:
use a sort-based deduplication or hash-based(flat_hash_map) deduplication algorithm instead of nested loop

Reply:
general imp for all types.